### PR TITLE
Migrated button to CSS modules

### LINKED ISSUE #573: Convert Button component to CSS modules
# Issue Description
To better align with modern CSS practices and modern CSS theming capabilities, we want to update this component to use CSS modules and if applicable, CSS custom properties.

## Issue Type

- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Component styles are converted to use CSS modules. If applicable, convert SCSS variables to CSS custom properties.

## Current Behavior
Component styles use manual name spaced classes and SCSS variables.

### PR DESCRIPTION
**Pull request by mhemesath**
### Summary
* Updated variables to custom properties
* Converted CSS to be a local module
* Updated tests

Resolves #573 

### Deployment Link
https://terra-core-deployed-pr-546.herokuapp.com/static/#/site/button

### CODE REVIEW

**Comment by mhemesath** [comment]
I'm having issues with the nightwatch tests. One thing I'm noticing is that they seem to duplicate our jest tests. For example

```js
  'Displays an icon only toggle-button': (browser) => {
    browser
      .url(`http://localhost:${browser.globals.webpackDevServerPort}/#/tests/toggle-button-tests/icon-only`)
      .assert.attributeContains('.button', 'aria-label', 'Custom Text')
      .assert.containsText('.button', '');
  },
```

I propose that we remove class name assertions from nightwatch and simply grab the elements by ID to query them. Thoughts?

**Comment by JakeLaCombe** [comment]
I tend to agree as well. I feel like making sure that specific attributes are set on an element should be the responsibility of our jest tests, not nightwatch tests

**Comment by bjankord** [comment]
@mhemesath @JakeLaCombe I like this proposal, I think this is a good topic for our meeting later today. I added a topic to discuss how we want to use nightwatch to test css module based components to today's meeting agenda.

**Comment by bjankord** [comment]
Update the changelog and then the build should pass. Here's what I added for badge changelog: https://github.com/cerner/terra-core/blob/master/packages/terra-badge/CHANGELOG.md

File: packages/terra-search-field/tests/nightwatch/search-field-spec.js [hunk 1]
```diff
@@ -73,8 +73,10 @@ module.exports = {
       .assert.elementPresent('.terra-SearchField-button')
       .assert.attributeEquals('.terra-SearchField-input', 'placeholder', '');
 
-    browser.getCssProperty('.terra-SearchField-input', 'height', (result) => {
-      browser.assert.cssProperty('.terra-SearchField-button', 'height', result.value);
+    browser.getCssProperty('.terra-SearchField-input', 'height', (inputResult) => {
+      browser.getCssProperty('.terra-SearchField-button', 'height', (buttonResult) => {
+        browser.assert.equal(Math.round(parseFloat(inputResult.value)), Math.round(parseFloat(buttonResult.value)));
```
Comment (emilyrohrbough) [inline packages/terra-search-field/tests/nightwatch/search-field-spec.js]: What do you have this change?

Comment (mhemesath) [inline packages/terra-search-field/tests/nightwatch/search-field-spec.js]: for some reason locally it was off by 0.011px. Rounding ensures that it passes if iwthin .5px

Comment (emilyrohrbough) [inline packages/terra-search-field/tests/nightwatch/search-field-spec.js]: So weird.

Comment (mhemesath) [inline packages/terra-search-field/tests/nightwatch/search-field-spec.js]: Yeah i visually examined it, and it looked fine. It was just slightly off. @Matt-Butler has said he's seen this before when comparing CSS prop values

**Comment by mmalaker** [comment]
Functional verification completed. Button examples display and function correctly after conversion to CSS modules.

**Pull request merged**